AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/UserPoolSignInHelper.swift

```diff
@@ -136,19 +114,27 @@ struct UserPoolSignInHelper: DefaultLogger {
                     parameters: parameters)
 
                 switch challengeName {
-                case .smsMfa, .customChallenge, .newPasswordRequired, .softwareTokenMfa, .selectMfaType:
+                case .smsMfa, .customChallenge, .newPasswordRequired, .softwareTokenMfa, .selectMfaType, .emailOtp:
                     return SignInEvent(eventType: .receivedChallenge(respondToAuthChallenge))
                 case .deviceSrpAuth:
                     return SignInEvent(eventType: .initiateDeviceSRP(username, response))
                 case .mfaSetup:
                     let allowedMFATypesForSetup = respondToAuthChallenge.getAllowedMFATypesForSetup
-                    if allowedMFATypesForSetup.contains(.totp) {
-                        return SignInEvent(eventType: .initiateTOTPSetup(username, response))
+                    if allowedMFATypesForSetup.contains(.totp) && allowedMFATypesForSetup.contains(.email) {
+                        return SignInEvent(eventType: .receivedChallenge(respondToAuthChallenge))
+                    } else if allowedMFATypesForSetup.contains(.totp) {
+                        return SignInEvent(eventType: .initiateTOTPSetup(username, respondToAuthChallenge))
+                    } else if allowedMFATypesForSetup.contains(.email) {
+                        return SignInEvent(eventType: .receivedChallenge(respondToAuthChallenge))
                     } else {
                         let message = "Cannot initiate MFA setup from available Types: \(allowedMFATypesForSetup)"
                         let error = SignInError.invalidServiceResponse(message: message)
                         return SignInEvent(eventType: .throwAuthError(error))
                     }
+                case.sdkUnknown(let challengeType):
```

Comment:
There is no difference between this  and the `default` case. 
Should we use `challengeType` somewhere?